src/app/core/store/organizations/contact/contacts.state.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {createEntityAdapter, EntityState} from "@ngrx/entity";
+import {ContactModel} from "./contact.model";
+import {createSelector} from "@ngrx/store";
+import {AppState} from "../../app.state";
+
+export interface ContactsState extends EntityState<ContactModel> {
+}
+
+export const contactsAdapter = createEntityAdapter<ContactModel>({selectId: contact => contact.code});
+
+export const initialContactsState: ContactsState = contactsAdapter.getInitialState({
+});
+
+export const selectContactsState = (state: AppState) => state.contacts;
+export const selectAllContacts = createSelector(selectContactsState, contactsAdapter.getSelectors().selectAll);
+export const selectContactByCode = (code) => createSelector(selectAllContacts, contacts => {
+  return contacts.find(contact => contact.code === code);
```

Comment:
I'm confused, because you set `organizationCode` here as parameter, but you use `contact.code` to find Contact. It means that contact code is made of organization code?

Reply:
Contact code is the code of organization for which the contact is. This is the binding attribute.

Reply:
I think we should use `id` instead of `code`, because `code` is not immutable attribute.